src/utils/auth_utils.jsx

```diff
@@ -0,0 +1,420 @@
+  const newUser = getAdditionalUserInfo(userCredential).isNewUser;
+  cookies.set(cookieKeys.ACCESS_TOKEN, auth.currentUser.accessToken, cookieConfig);
+  if (newUser) {
+    await createUserInDB(auth.currentUser.email, userCredential.user.uid, VOLUNTEER_ROLE, true);
+    cookies.set(cookieKeys.ROLE, VOLUNTEER_ROLE, cookieConfig);
+    navigate(newUserRedirectPath);
+  } else {
+    const user = await NPOBackend.get(`/users/${auth.currentUser.uid}`);
+    cookies.set(cookieKeys.ROLE, user.data.user.role, cookieConfig);
+    if (!user.data.user.registered) {
+      navigate(newUserRedirectPath);
+    } else {
+      navigate(defaultRedirectPath);
+    }
+  }
+};
+
+/**
+ * When a user signs in with Google for the first time, they will need to add additional info
+ * This is called when the user submits the additional information which will lead to the flag
+ * in the backend changed so that user is not new anymore
+ * @param {string} redirectPath path to redirect user
+ * @param {hook} navigate used to redirect the user after submitted
+ */
+const finishGoogleLoginRegistration = async (redirectPath, navigate) => {
+  // TODO: Switch to correct update route
+  await NPOBackend.put(`/users/update/${auth.currentUser.uid}`, {
+    registered: true,
+  });
+  navigate(redirectPath);
+};
+
+/**
+ * Logs a user in with email and password
+ * @param {string} email The email to log in with
+ * @param {string} password The password to log in with
+ * @param {string} redirectPath The path to redirect the user to after logging out
+ * @param {hook} navigate An instance of the useNavigate hook from react-router-dom
+ * @param {Cookies} cookies The user's cookies to populate
+ * @returns A boolean indicating whether or not the log in was successful
+ */
+const logInWithEmailAndPassword = async (email, password, redirectPath, navigate, cookies) => {
+  await signInWithEmailAndPassword(auth, email, password);
+  // Check if the user has verified their email.
+  if (!auth.currentUser.emailVerified) {
+    throw new Error('Please verify your email before logging in.');
+  }
+  cookies.set(cookieKeys.ACCESS_TOKEN, auth.currentUser.accessToken, cookieConfig);
+  const user = await NPOBackend.get(`/users/${auth.currentUser.uid}`);
+  console.log('Current user: ');
+  console.table(user.data);
+  cookies.set(cookieKeys.ROLE, user.data.role, cookieConfig);
+  navigate(redirectPath);
+};
+
+/**
+ * Creates a user in firebase database
+ * @param {string} email
+ * @param {string} password
+ * @returns A UserCredential object from firebase
+ */
+const createUserInFirebase = async (email, password) => {
+  const user = await createUserWithEmailAndPassword(auth, email, password);
+  return user.user;
+};
+
+/**
+ * Creates a user (both in firebase and database)
+ * @param {string} email
+ * @param {string} password
+ * @param {string} role
+ * @returns A UserCredential object from firebase
+ */
+const createUser = async (email, password, firstName, lastName, role) => {
```

Comment:
⚠️ **[eslint]** <[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'createUser' is assigned a value but never used.